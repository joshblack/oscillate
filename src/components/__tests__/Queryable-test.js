/* @noflow */

import React from 'react';
import {renderToString} from 'react-dom/server';

describe('Queryable HOC', () => {
  let fetch;
  let Provider;
  let Environment;
  let Queryable;
  let Feed;
  let env;

  beforeEach(() => {
    fetch = require('fbjs/lib/fetchWithRetries');
    Provider = require('../EnvironmentProvider').default;
    Environment = require('../../environment/Environment').default;
    Queryable = require('../Queryable').default;

    /* eslint-disable react/prop-types */
    class FeedComponent extends React.Component {
      render() {
        return (
          <div>
            {this.props.data.map((item, i) => <div key={i}>{item.name}</div>)}
          </div>
        );
      }
    }
    /* eslint-enable react/prop-types */

    Feed = Queryable(FeedComponent, {
      path: '/api/items',
    });
    env = new Environment();
  });

  it('should renderToString with the given data', async () => {
    const App = () => (
      <Provider environment={env}>
        <Feed />
      </Provider>
    );

    const promise = env.preload();

    fetch.mock.deferreds[0].resolve({
      status: 200,
      json: () =>
        Promise.resolve([{name: 'item 1'}, {name: 'item 2'}, {name: 'item 3'}]),
    });

    await promise;

    expect(renderToString(<App />)).toMatchSnapshot();
  });
});
