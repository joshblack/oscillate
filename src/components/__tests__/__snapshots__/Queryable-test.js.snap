// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queryable HOC should renderToString with the given data 1`] = `"<div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"1674719421\\"><div data-reactid=\\"2\\">item 1</div><div data-reactid=\\"3\\">item 2</div><div data-reactid=\\"4\\">item 3</div></div>"`;
